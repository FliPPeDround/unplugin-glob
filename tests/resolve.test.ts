import { describe, expect, test } from 'vitest'
import { rollup } from 'rollup'
import glob from 'fast-glob'
import Plugin from '../src/rollup'

describe('resolve', async () => {
  const fixtures = await glob('./fixtures/*', {
    absolute: true,
    cwd: __dirname,
  })
  for (const fixture of fixtures) {
    test(fixture, async () => {
      const bundle = await rollup({
        input: fixture,
        plugins: [Plugin()],
      })
      const { output } = await bundle.generate({})
      expect(output[0].code).toMatchSnapshot()
    })
  }
})
