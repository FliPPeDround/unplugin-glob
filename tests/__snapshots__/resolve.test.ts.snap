// Vitest Snapshot v1

exports[`resolve 1`] = `
[
  "export { a, f as foo } from './b-2e2c7b43.js';
",
  "export { a, f as foo } from './b-2e2c7b43.js';
",
  "export { a, f as foo } from './b-2e2c7b43.js';
",
  "const a = 'a';

const foo = 'bar';

export { a, foo as f };
",
]
`;
